test/extended/operators/olm.go

```diff
@@ -70,4 +70,30 @@ var _ = g.Describe("[Feature:Platform] OLM should", func() {
 			}
 		})
 	}
+
+	// OCP-20981, [BZ 1626434]The olm/catalog binary should output the exact version info
+	// author: jiazha@redhat.com
+	g.It("olm version should contain the source commit id", func() {
+		catalogPodName, err := oc.AsAdmin().Run("get").Args("-n", "openshift-operator-lifecycle-manager", "pods", "-l", "app=catalog-operator", "-o=jsonpath={.items[0].metadata.name}").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		e2e.Logf("catalog-operator pod name:%s", catalogPodName)
+
+		g.By("get olm version from the catalog-operator pod")
+		oc.SetNamespace("openshift-operator-lifecycle-manager")
+		commands := []string{"exec", catalogPodName, "--", "olm", "--version"}
+		olmVersion, err := oc.AsAdmin().Run(commands...).Args().OutputToFile("version.json")
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("get the commit id from the version info")
+		gitCommitID, err := exec.Command("cat", olmVersion, "|", "grep", "'git commit'", "|", "awk", "'{print $3}'").CombinedOutput()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		e2e.Logf("olm version is:%s", string(gitCommitID))
+
+		g.By("get all commits from the operator-lifecycle-manager repo")
+		commitIDs, err := exec.Command("curl", "-k", "https://github.com/operator-framework/operator-lifecycle-manager/commits/master").CombinedOutput()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(string(commitIDs)).To(o.ContainSubstring(string(gitCommitID)))
```

Comment:
olm outputs the full commit sha, not the short. Can we assert on the length to ensure it's the full commit id?

Reply:
Thanks! It would be better. Do you have any ideas?

Reply:
we can check for the length, it should be 40-character long
```
len(gitCommitID) == 40
``

Reply:
Ok, thanks! Done.